js/adapt-stateful-session.js

```diff
@@ -125,12 +128,34 @@ define([
 			this.saveSessionState();
 		},
 
-		onCompletion: function() {
-			if (!this.checkTrackingCriteriaMet()) return;
-
+		onTrackingComplete: function(completionData) {
 			this.saveSessionState();
 			
-			Adapt.offlineStorage.set("status", this._config._reporting._onTrackingCriteriaMet);
+			var completionStatus = completionData.status.asLowerCase;
+
+			// The config allows the user to override the completion state.
+			switch (completionData.status) {
+				case COMPLETION_STATE.COMPLETED:
+				case COMPLETION_STATE.PASSED: {
+					if (!this._config._reporting._onTrackingCriteriaMet) {
+						Adapt.log.warn("No value defined for '_onTrackingCriteriaMet', so defaulting to '" + completionStatus + "'");
+					} else {
+						completionStatus = this._config._reporting._onTrackingCriteriaMet;
+					}
+
+					break;
+				}
+
+				case COMPLETION_STATE.FAILED: {
+					if (!this._config._reporting._onAssessmentFailure) {
+						Adapt.log.warn("No value defined for '_onAssessmentFailure', so defaulting to '" + completionStatus + "'");
+					} else {
+						completionStatus = this._config._reporting._onAssessmentFailure;
+					}
+				}
+			}
+
+			Adapt.offlineStorage.set("status", completionData.status.asLowerCase);
```

Comment:
you should be using `completionStatus` here not `completionData.status.asLowerCase`